component/componenterr/errors.go

```diff
@@ -14,13 +14,25 @@
 
 // Package oterr provides helper functions to create and process
 // OpenTelemetry specific errors
-package oterr
+package componenterr
```

Comment:
Maybe this?
```suggestion
package err
```

Reply:
Look at bad package names https://blog.golang.org/package-names. "err" is too generic.